public/index.html

```diff
@@ -22,6 +22,13 @@
       <p>Please enable JavaScript to continue.</p>
       <p>Powered by <a href="https://getodk.org">ODK</a>.</p>
     </noscript>
+    <script>
+      // We used to have hash-based navigation, we have now switched to web-history-based
+      // navigation. To support, bookmarked links, we are redirecting old URL to the new one.
+      if (window.location.href.startsWith(window.location.origin + "/#")) {
+          window.location.replace(window.location.origin + window.location.hash.substring(1));
```

Comment:
Is there an advantage in doing this here rather than somewhere else in the app? I was wondering about a couple of other possibilities:

**Idea 1**

Could we do this in src/main.js instead? Before Vue mounts the app, main.js could complete this check and redirect if need be. I kind of like the idea of keeping public/index.html as simple as possible and free of JavaScript.

**Idea 2**

Maybe we could even do this in Vue Router? Would it be preferable to avoid a true redirect?

```js
router.beforeEach(to => (to.path === '/' && to.hash.startsWith('/')
  ? to.hash
  : true));
```

Reply:
went with the idea 2